File: ruby/3b.rb
#!/usr/bin/ruby

# Convert input into an array of arrays of side lengths, first transposing so that sides are grouped vertically
input = File.open('3.in').readlines.map{|x| x.split.map(&:to_i)}.transpose.flatten.each_slice(3).to_a

# Select the triangles whose border length less the maximum is greater than the maximum
puts input.select{|t| t.reduce(:+) - t.max > t.max}.size

File: python/3b.py
#!/usr/bin/python

import itertools

def main():
    # Process input into a list of lists
    data = [[int(y) for y in x.split()] for x in open('3.in')]

    # Transpose
    data = zip(*data)

    # Flatten
    data = list(itertools.chain(*data))

    # Chunk into groups of 3 sorted
    data = [sorted(data[x:x+3], reverse=True) for x in range(0, len(data), 3)]

    # Select triangles whose border less the largest side is larger than the largest side
    valid = [x for x in data if sum(x) - x[0] > x[0]]

    # Answer is the number of valid triangles
    print len(valid)

if __name__ == '__main__':
    main()

# vi: ts=4 et:
